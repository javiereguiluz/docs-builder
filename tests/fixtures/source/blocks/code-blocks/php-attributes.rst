.. code-block:: php-attributes

    // src/SomePath/SomeClass.php
    namespace App\SomePath;

    use Symfony\Component\Validator\Constraints as Assert;

    class SomeClass
    {
        #[AttributeName]
        private $property1;

        #[AttributeName()]
        private $property2;

        #[AttributeName('value')]
        private $property3;

        #[AttributeName('value', option: 'value')]
        private $property4;

        #[AttributeName(['value' => 'value'])]
        private $property5;

        #[AttributeName(
            'value',
            option: 'value'
        )]
        private $property6;

        #[Assert\AttributeName('value')]
        private $property7;

         #[Assert\AttributeName(
            'value',
            option: 'value'
         )]
         private $property8;

         #[Route('/blog/{page<\d+>}', name: 'blog_list')]
         private $property9;

         #[Assert\GreaterThanOrEqual(
             value: 18,
         )]
         private $property10;

         #[ORM\CustomIdGenerator(class: 'doctrine.uuid_generator')]
         private $property11;
    }

