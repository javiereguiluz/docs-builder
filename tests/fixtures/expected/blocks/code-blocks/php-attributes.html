<div translate="no" data-loc="45" class="notranslate codeblock codeblock-length-md codeblock-php-attributes codeblock-php">
    <div class="codeblock-scroll">
        <pre class="codeblock-lines">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45</pre>
        <pre class="codeblock-code">
            <code>
                <span class="hljs-comment">// src/SomePath/SomeClass.php</span>
<span class="hljs-keyword">namespace</span> <span class="hljs-title">App</span>\<span class="hljs-title">SomePath</span>;
                <span class="hljs-keyword">use</span><span class="hljs-title">Symfony</span>\<span class="hljs-title">Component</span>\<span class="hljs-title">Validator</span>\<span class="hljs-title">Constraints</span> <span class="hljs-title">as</span> <span class="hljs-title">Assert</span>;
                <span class="hljs-class">
                    <span class="hljs-keyword">class</span>
                    <span class="hljs-title">SomeClass</span>
                </span>
                {
                <span class="hljs-php-attribute">#[AttributeName]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                    <span class="hljs-variable-other-marker">$</span> property1</span>
                ;
                <span class="hljs-php-attribute">#[AttributeName(</span>
                <span class="hljs-php-attribute">)]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                    <span class="hljs-variable-other-marker">$</span> property2</span>
                ;
                <span class="hljs-php-attribute">#[AttributeName(</span>
                <span class="hljs-string">'value'</span>
                <span class="hljs-php-attribute">)]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                    <span class="hljs-variable-other-marker">$</span> property3</span>
                ;
                <span class="hljs-php-attribute">#[AttributeName(</span>
                <span class="hljs-string">'value'</span>
                , option:
                <span class="hljs-string">'value'</span>
                <span class="hljs-php-attribute">)]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                    <span class="hljs-variable-other-marker">$</span> property4</span>
                ;
                <span class="hljs-php-attribute">#[AttributeName(</span>
[<span class="hljs-string">'value'</span> =&gt; <span class="hljs-string">'value'</span>]<span class="hljs-php-attribute">)]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                    <span class="hljs-variable-other-marker">$</span> property5</span>
                ;
                <span class="hljs-php-attribute">#[AttributeName(</span>
                <span class="hljs-string">'value'</span>
                , option:
                <span class="hljs-string">'value'</span>
                <span class="hljs-php-attribute">)]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                    <span class="hljs-variable-other-marker">$</span> property6</span>
                ;
                <span class="hljs-php-attribute">#[Assert\AttributeName(</span>
                <span class="hljs-string">'value'</span>
                <span class="hljs-php-attribute">)]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                    <span class="hljs-variable-other-marker">$</span> property7</span>
                ;
                <span class="hljs-php-attribute">#[Assert\AttributeName(</span>
                <span class="hljs-string">'value'</span>
                , option:
                <span class="hljs-string">'value'</span>
                <span class="hljs-php-attribute">)]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                    <span class="hljs-variable-other-marker">$</span> property8</span>
                ;
                <span class="hljs-php-attribute">#[Route(</span>
                <span class="hljs-string">'/blog/{page&lt;\d+&gt;}'</span>
                , name:
                <span class="hljs-string">'blog_list'</span>
                <span class="hljs-php-attribute">)]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                <span class="hljs-variable-other-marker">$</span> property9</span>
                ;
                <span class="hljs-php-attribute">#[Assert\GreaterThanOrEqual(</span>
                value:
                <span class="hljs-number">18</span>
                ,
                <span class="hljs-php-attribute">)]</span>
                <span class="hljs-keyword">private</span>
                <span class="hljs-variable">
                <span class="hljs-variable-other-marker">$</span> property10</span>
                ;
}</code></pre>
    </div>
</div>
